packages/design-system/src/components/primitives/use-scrub.ts

```diff
@@ -0,0 +1,83 @@
+import { useEffect, useRef } from "react";
+import {
+  numericScrubControl,
+  type NumericScrubValue,
+} from "./numeric-gesture-control";
+import { unstable_batchedUpdates as unstableBatchedUpdates } from "react-dom";
+
+export const useScrub = ({
+  value,
+  onChange,
+  onChangeComplete,
+  shouldHandleEvent,
+}: {
+  value: NumericScrubValue;
+  onChange: (value: NumericScrubValue) => void;
+  onChangeComplete?: (value: NumericScrubValue) => void;
+  shouldHandleEvent?: (node: EventTarget) => boolean;
+}) => {
+  const scrubRef = useRef<HTMLDivElement | null>(null);
```

Comment:
It looks to me that it's not needed at all, and we can use `inputRef.parent.appendChild(<div tabindex=-1)` whenever needed. The only thing it used is to preserve focus inside input container

Reply:
its also used for focus inside css value input ` scrubRef.current?.focus();`

Reply:
For same reason, to preserve focus inside container